### DISCUSSION TITLE
resample generating RTs in a deterministic manner

### DISCUSSION THREAD

**Discussion by yanivabir** [mwshinn/PyDDM#50] [Ask for help] [answered]
Hi,

I'm trying to sample fake data from a simple model where drift depends linearly on coherence. I'm sampling in the most straightforward and perhaps stupid way I could think of, and I'm getting some weird data.

You can find everything in this colab: https://colab.research.google.com/drive/12Y6XF1YKUx9iYjH0TY1atpkKitA0X4T2?usp=sharing

But basically, this is the model:
```
class DriftCoherence(ddm.models.Drift):
    name = "Drift depends linearly on uncertainty"
    required_parameters = ["driftcoh"] 
    required_conditions = ["coh"] 

    # We must always define the get_drift function, which is used to compute the instantaneous value of drift.
    def get_drift(self, conditions, **kwargs):
        return self.driftcoh * conditions['coh']

model = Model(name='Simple model',
              drift=DriftCoherence(driftcoh=0.77),
              noise=NoiseConstant(noise=1),
              bound=BoundConstant(B=2),
              overlay=OverlayNonDecision(nondectime=.2),
              dx=.001, dt=.01, T_dur=5)
```

This is how I sample:
```
cohs = np.arange(0.0, 0.6, step = .005)
for i in range(cohs.shape[0]):
  if i == 0:
    smp = model.solve(conditions={"coh": cohs[i]}).resample(1)
  else:
    smp += model.solve(conditions={"coh": cohs[i]}).resample(1)

smpdf = smp.to_pandas_dataframe()
```

And plotting RTs against `coh`, I get a deterministic and weird function:
![index](https://user-images.githubusercontent.com/16220995/149817608-1f596556-c5d2-4030-ae1a-8b4690d3efbe.png)
 
Here's a histogram as well:
![index2](https://user-images.githubusercontent.com/16220995/149817683-7d9750b4-6629-4444-aa6c-20f19a35ade2.png)

Any ideas why this would happen?

Thanks!
Yaniv

**Comment by mwshinn** [comment, marked as answer]
Looks like the problem here is that you're using the same seed for each sample.  I agree this is confusing so I updated the code to not use a seed if no seed is specified (see commit 8c46da227ae217a96118004661a919058559a039).  For a workaround now, you can add "seed=i" as an argument to the "resample" call.

**Reply by yanivabir** [reply]
Thanks! that solved it.
